rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1039,23 +1036,99 @@ public int getBtcBlockchainInitialBlockHeight() throws IOException {
         int maxDepth = head.getHeight() - getLowestBlock().getHeight();
 
         if (depth < 0 || depth > maxDepth) {
             throw new IndexOutOfBoundsException(String.format("Depth must be between 0 and %d", maxDepth));
         }
 
-        int currentDepth = 0;
-        StoredBlock current = head;
-        while (currentDepth < depth) {
-            current = current.getPrev(btcBlockStore);
-            currentDepth++;
+        StoredBlock blockAtDepth = btcBlockStore.getStoredBlockAtMainChainDepth(depth);
+        return blockAtDepth.getHeader().getHash();
+    }
+
+    public Long getBtcTransactionConfirmationsGetCost(Sha256Hash btcBlockHash) {
+        final long BASIC_COST = 13_600;
+        final long STEP_COST = 70;
+
+        // Dynamic cost based on the depth of the block that contains
+        // the transaction. Find such depth first, then calculate
+        // the cost.
+        Context.propagate(btcContext);
+        try {
+            this.ensureBtcBlockStore();
+            final StoredBlock block = btcBlockStore.getFromCache(btcBlockHash);
+
+            // Block not found, default to basic cost
+            if (block == null) {
+                return BASIC_COST;
+            }
+
+            final int bestChainHeight = getBtcBlockchainBestChainHeight();
+
+            // Make sure calculated depth is >= 0
+            final int blockDepth = Math.max(0, bestChainHeight - block.getHeight());
+
+            // Block too deep, default to basic cost
+            if (blockDepth > BTC_TRANSACTION_CONFIRMATION_MAX_DEPTH) {
+                return BASIC_COST;
+            }
+
+            return BASIC_COST + blockDepth*STEP_COST;
+        } catch (IOException | BlockStoreException e) {
+            logger.warn("getBtcTransactionConfirmationsGetCost btcBlockHash:{} there was a problem " +
+                    "gathering the block depth while calculating the gas cost. " +
+                    "Defaulting to basic cost.", btcBlockHash, e);
+            return BASIC_COST;
+        }
+    }
+
+    /**
+     * @param btcTxHash The BTC transaction Hash
+     * @param btcBlockHash The BTC block hash
+     * @param merkleBranch The merkle branch
+     * @throws BlockStoreException
+     * @throws IOException
+     */
+    public Integer getBtcTransactionConfirmations(Sha256Hash btcTxHash, Sha256Hash btcBlockHash, MerkleBranch merkleBranch) throws BlockStoreException, IOException {
+        Context.propagate(btcContext);
+        this.ensureBtcBlockChain();
+
+        // Get the block using the given block hash
+        StoredBlock block = btcBlockStore.getFromCache(btcBlockHash);
+        if (block == null) {
+            return BTC_TRANSACTION_CONFIRMATION_INEXISTENT_BLOCK_HASH_ERROR_CODE;
+        }
+
+        final int bestChainHeight = getBtcBlockchainBestChainHeight();
+
+        // Prevent diving too deep in the blockchain to avoid high processing costs
+        int blockDepth = bestChainHeight - block.getHeight();
```

Comment:
In `getCost` you do ` final int blockDepth = Math.max(0, bestChainHeight - block.getHeight());` but not here.

Reply:
good catch